### PR TITLE
Add a few minimal debugging docs.

### PR DESCRIPTION
**Pull request by emilio**
r? @jdm (or anyone else)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14187)

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+You can also, of course, run directly your debugger on the Servo binary:
+
+```
+$ gdb --args ./target/debug/servo test.html
+```
+
+## Debugging SpiderMonkey.
+
+You can build Servo with a debug version of SpiderMonkey passing the
+`--debug-mozjs` flag to `mach`.
```
Comment (jdm) [inline docs/debugging.md]: `./mach build`

File: docs/debugging.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Note that this sometimes can cause problems when an existing build exists, so
+you might have to delete the `mozjs` build directory, or running `./mach clean`.
```
Comment (jdm) [inline docs/debugging.md]:
```
or run `./mach clean` before performing the first `--debug-mozjs` build.
```

**Review by jdm — CHANGES_REQUESTED** [review]
This is a nice improvement over the status quo of no docs :)

**Comment by emilio** [comment]
Amended, let me know if you think it's good to go :)

**Pull request merged**